core/src/main/scala/org/apache/spark/api/java/JavaRDDLike.scala

```diff
@@ -271,6 +271,15 @@ trait JavaRDDLike[T, This <: JavaRDDLike[T, This]] extends Serializable {
     rdd.pipe(command.asScala, env.asScala)
 
   /**
+   * Return an RDD created by piping elements to a forked external process.
+   */
+  def pipe(command: JList[String],
+           env: java.util.Map[String, String],
```

Comment:
Okay, I changed to use JMap and JIterator across the file. 

Regarding the arguments for `printPipeContext` and `printRDDElement`, simply passing `VoidFunction[String]` is not sufficient because `printPipeContext`  is of type ` (String => Unit) => Unit`.  Any idea how to deal with that?